### PR TITLE
24hs volume

### PR DESCRIPTION
**Pull request by hpmaxi**
This calculates the collateral volume of the last day.

### CODE REVIEW

File: app/src/components/market/market_view_container.tsx [hunk 1]
```diff
@@ -30,6 +74,18 @@ const MarketViewContainer: React.FC<Props> = (props: Props) => {
     resolution,
   } = marketMakerData
 
+  useEffect(() => {
+    const get24hsVolume = async () => {
+      const BLOCKS_PER_SECOND = 15
+      const OFFSET = (60 * 60 * 24) / BLOCKS_PER_SECOND
```
Comment (fvictorio) [inline app/src/components/market/market_view_container.tsx]: Add `Math.round`

File: app/src/components/market/market_view_container.tsx [hunk 2]
```diff
@@ -1,21 +1,65 @@
...
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  const { data: volumeBefore, error: errorVolumeBefore } = useQuery(buildQuery24hsEarlier(hash && hash.toLowerCase()), {
+    skip: !!lastDayVolume,
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  if (errorVolumeBefore || errorVolumeNow) {
+    setLastDayVolume('-')
```
Comment (fvictorio) [inline app/src/components/market/market_view_container.tsx]: Log error.

File: app/src/components/market/market_view_container.tsx [hunk 3]
```diff
@@ -1,21 +1,65 @@
...
+  const { library: provider } = context
+
+  const [lastDayVolume, setLastDayVolume] = useState<Maybe<string>>(null)
+
+  const { data: volumeNow, error: errorVolumeNow } = useQuery(GET_COLLATERAL_VOLUME_NOW, {
+    skip: !!lastDayVolume,
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  const { data: volumeBefore, error: errorVolumeBefore } = useQuery(buildQuery24hsEarlier(hash && hash.toLowerCase()), {
```
Comment (fvictorio) [inline app/src/components/market/market_view_container.tsx]: Does this mean that while `hash` is null, the query will be performed with `null`? Doesn't this cause any problems?

File: app/src/components/market/market_view_container.tsx [hunk 4]
```diff
@@ -1,21 +1,65 @@
...
+    variables: { id: marketMakerAddress.toLowerCase() },
+  })
+
+  if (errorVolumeBefore || errorVolumeNow) {
+    setLastDayVolume('-')
+  } else if (volumeNow && volumeBefore) {
+    const now = new BigNumber(volumeNow.fixedProductMarketMakers[0].collateralVolume)
+    const before = new BigNumber(volumeBefore.fixedProductMarketMakers[0].collateralVolume)
+
+    setLastDayVolume(now.sub(before).toString())
```
Comment (fvictorio) [inline app/src/components/market/market_view_container.tsx]: I think `lastDayVolume` should be a BigNumber. Converting it to string is a view responsibility.

File: app/src/components/market/market_view_container.tsx (hunk 3 above)
Comment (fvictorio) [reply app/src/components/market/market_view_container.tsx]: `skip` can be used here, right? Is using `skip: !!lastDayVolume || !hash` below a good solution?

Comment (hpmaxi) [reply app/src/components/market/market_view_container.tsx]: Yes, seems like a good solution to me.

**Pull request merged**